File: Problem 67.py
matrix = []
with open("Problem 67 Tree.txt") as file:
    for line in file:
        matrix.append([int(x) for x in line.split(" ")])
new = [[0 for _ in line] for line in matrix]
new[0][0] = matrix[0][0]

def printMatrix(toPrint):
    for line in toPrint:
        thing = ""
        for n in line:
            if n < 10:
                thing += "0"
            thing += str(n)+" "
        print(thing[:-1])

for y in range(len(matrix)-1):
    for x in range(len(matrix[y])):
        new[y+1][x] = max(new[y][x]+matrix[y+1][x],new[y+1][x])
        new[y+1][x+1] = max(new[y][x]+matrix[y+1][x+1],new[y+1][x+1])

print(max(new[-1]))

File: Problem 67.hs
import Data.Graph
import Data.Tree

combineLists xs@(x:_) (y:ys) = (x+y) : combineLists' xs ys
	where	combineLists' (x:[]) (y:[]) = [x+y]
		combineLists' (x1:x2:xs) (y:ys) = (max (x1+y) (x2+y)) : combineLists' (x2:xs) ys

main = readFile "Problem 67 Tree.txt" >>= print . maximum . foldl1 combineLists . map (map read . words) . lines